src/features/user/UserProfile/components/EmbedModal.tsx

```diff
@@ -0,0 +1,169 @@
+import { Modal, Snackbar } from '@material-ui/core'
+import React, { ReactElement } from 'react'
+import ToggleSwitch from '../../../common/InputTypes/ToggleSwitch';
+import styles from '../styles/EmbedModal.module.scss';
+import i18next from '../../../../../i18n';
+import { useAuth0 } from '@auth0/auth0-react';
+import { putAuthenticatedRequest } from '../../../../utils/apiRequests/api';
+import { useRouter } from 'next/router';
+import MuiAlert from '@material-ui/lab/Alert';
+import BackButton from '../../../../../public/assets/images/icons/BackButton';
+
+interface Props {
+    embedModalOpen: boolean;
+    setEmbedModalOpen: Function;
+    userprofile: Object;
+}
+
+const { useTranslation } = i18next;
+
+export default function EmbedModal({ embedModalOpen, setEmbedModalOpen, userprofile }: Props) {
+
+    const { t, ready } = useTranslation(['editProfile']);
+
+    const [isPrivate, setIsPrivate] = React.useState(false);
+    const [isUploadingData, setIsUploadingData] = React.useState(false);
+    const [token, setToken] = React.useState('');
+    const [severity, setSeverity] = React.useState('success')
+    const [snackbarMessage, setSnackbarMessage] = React.useState("OK");
+    const [snackbarOpen, setSnackbarOpen] = React.useState(false);
+    const router = useRouter();
+    const {
+        isLoading,
+        isAuthenticated,
+        getAccessTokenSilently
+    } = useAuth0();
+    // This effect is used to get and update UserInfo if the isAuthenticated changes
+    React.useEffect(() => {
+        async function loadFunction() {
+            const token = await getAccessTokenSilently();
+            setToken(token);
+        }
+        if (isAuthenticated && !isLoading) {
+            loadFunction()
+        }
+    }, [isAuthenticated, isLoading])
+
+    React.useEffect(() => {
+        if (userprofile && userprofile.isPrivate) {
+            setIsPrivate(true);
+        }
+    }, [userprofile]);
+
+    const handleToggleChange = (e: any) => {
+        setIsPrivate(!e.target.checked);
+    }
+
+    const handleSnackbarOpen = () => {
+        setSnackbarOpen(true);
+    };
+    const handleSnackbarClose = (
+        event?: React.SyntheticEvent,
+        reason?: string
+    ) => {
+        if (reason === 'clickaway') {
+            return;
+        }
+        setSnackbarOpen(false);
+    };
+
+    const saveProfile = async () => {
+        setIsUploadingData(true);
+        const bodyToSend = {
+            isPrivate: false
+        }
+        if (!isLoading && token) {
+            try {
+                putAuthenticatedRequest(`/app/profile`, bodyToSend, token).then((res) => {
+                    setSeverity('success')
+                    setSnackbarMessage(ready ? t('editProfile:profileSaved') : '')
+                    setEmbedModalOpen(false)
+                    setIsUploadingData(false)
+                    router.push(`${process.env.WIDGET_URL}?user=${userprofile.id}&tenantkey=${process.env.TENANTID}`);
```

Comment:
There is a bug in our TreeMap and also in our TreeProfile widget that it is crashing if a `TENANTID` is given which does not exist in https://github.com/Plant-for-the-Planet-org/planet-widgets/blob/develop/tenant.json
e.g. https://widgets.plant-for-the-planet.org/?user=prf_KL79s31SQaotgohmtXiETjcq&tenantkey=ten_NxJq55pm

As we cannot always add all possible tenants in the widgets and wrong `tenantkey` parameters can be specified, the widget code must be failsafe for non existing `tenantkey` parameters given!

Reply:
I added a fix for the widgets with https://github.com/Plant-for-the-Planet-org/planet-widgets/pull/49

Reply:
I merged the PR on the widgets repo

Reply:
Also must be released before we release this :-)